RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/PixelClustering.h

```diff
@@ -568,15 +570,25 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE::pixelClustering {
 #endif
 
         [[maybe_unused]] const uint32_t blockDimension = alpaka::getWorkDiv<alpaka::Block, alpaka::Elems>(acc)[0u];
-        // assume that we can cover the whole module with up to maxIterGPU blockDimension-wide iterations
-        ALPAKA_ASSERT_ACC((hist.size() / blockDimension) < maxIterGPU);
+        // assume that we can cover the whole module with up to maxIterClustering blockDimension-wide iterations
+        ALPAKA_ASSERT_ACC((hist.size() / blockDimension) < TrackerTraits::maxIterClustering);
 
         // number of elements per thread
-        constexpr uint32_t maxElements =
-            cms::alpakatools::requires_single_thread_per_block_v<Acc1D> ? maxElementsPerBlock : 1;
+        const uint32_t maxElements = cms::alpakatools::requires_single_thread_per_block_v<Acc1D>
+                                         ? (enableDigiMorphing ? maxElementsPerBlockMorph : maxElementsPerBlock)
+                                         : 1;
+
+#ifdef GPU_DEBUG
+        const auto nthreads = alpaka::getWorkDiv<alpaka::Thread, alpaka::Elems>(acc)[0u];
```

Comment:
Mhm, this check is confusing to me 🤔 

`alpaka::getWorkDiv<alpaka::Thread, alpaka::Elems>(acc)[0u]` is _not_ the number of threads, it is the number of elements per threads, which is expected to be 1 on GPU and correspond to the block size on CPU.